### PR TITLE
[Selenium] Fix unstable selenium tests from dashboard package

### PR DESCRIPTION
**Pull request by SkorikSergey**
### What does this PR do?
This PR fixes unstable selenium tests from ```Dashboard``` and `Factories` packages:
1. Check that required devfile is selected in devfiles list before starting a new workspace (it may affect on all selenium tests that create workspace from `Dashboard`)
2. Check that selenium tests are correctly switched to `Theia IDE` (it may affect on all selenium tests that start workspaces from `Dashboard`).

### CODE REVIEW

**Comment by SkorikSergey** [comment]
ci-test

**Comment by SkorikSergey** [comment]
ci-test

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/theia/TheiaIde.java [hunk 1]
```diff
@@ -151,7 +153,11 @@ public void waitNotificationPanelClosed() {
   }
 
   public void waitTheiaIde() {
-    seleniumWebDriverHelper.waitVisibility(theiaIde, PREPARING_WS_TIMEOUT_SEC);
+    try {
+      seleniumWebDriverHelper.waitVisibility(theiaIde, PREPARING_WS_TIMEOUT_SEC);
+    } catch (WebDriverException ex) {
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/theia/TheiaIde.java]: Can we be more specific here, not to cover unrecoverable exceptions, but timeout exception?

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/NewWorkspace.java [hunk 2]
```diff
@@ -185,7 +186,12 @@ public boolean isDevfileVisible(Devfile devfile) {
 
   public void selectDevfile(Devfile devfile) {
     waitDevfiles(asList(devfile));
-    seleniumWebDriverHelper.waitAndClick(By.xpath(format(DEVFILE_ROW_XPATH, devfile.getId())));
+    try {
+      seleniumWebDriverHelper.waitAndClick(By.xpath(format(DEVFILE_ROW_XPATH, devfile.getId())));
+      waitDevfileSelected(devfile);
+    } catch (WebDriverException ex) {
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/NewWorkspace.java]: Can we be more specific here, not to cover unrecoverable exceptions, but timeout exception?

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/DirectUrlFactoryWithKeepDirectoryTest.java [hunk 3]
```diff
@@ -71,7 +71,7 @@ public void factoryWithDirectUrlWithKeepDirectory() {
     theiaIde.switchToIdeFrame();
     theiaIde.waitTheiaIde();
     theiaIde.waitLoaderInvisibility();
-    theiaIde.waitNotificationEqualsTo("Che Workspace: Finished cloning projects.");
+    theiaIde.waitNotificationDisappearance("Che Workspace: Finished cloning projects.");
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/DirectUrlFactoryWithKeepDirectoryTest.java]: It duplicates next waitNotificationDisappearance().

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/DirectUrlFactoryWithRootFolderTest.java [hunk 4]
```diff
@@ -90,7 +90,7 @@ public void factoryWithDirectUrlWithRootFolder() {
     theiaIde.switchToIdeFrame();
     theiaIde.waitTheiaIde();
     theiaIde.waitLoaderInvisibility();
-    theiaIde.waitNotificationEqualsTo("Che Workspace: Finished cloning projects.");
+    theiaIde.waitNotificationDisappearance("Che Workspace: Finished cloning projects.");
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/DirectUrlFactoryWithRootFolderTest.java]: It duplicates next waitNotificationDisappearance().

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/DirectUrlFactoryWithSpecificBranchTest.java [hunk 5]
```diff
@@ -90,7 +90,7 @@ public void factoryWithDirectUrlWithSpecificBranch() {
     theiaIde.switchToIdeFrame();
     theiaIde.waitTheiaIde();
     theiaIde.waitLoaderInvisibility();
-    theiaIde.waitNotificationEqualsTo("Che Workspace: Finished cloning projects.");
+    theiaIde.waitNotificationDisappearance("Che Workspace: Finished cloning projects.");
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/DirectUrlFactoryWithSpecificBranchTest.java]: It duplicates next waitNotificationDisappearance().

**Comment by dmytro-ndp** [comment]
Could you, please, add list of tests in description of PR, and concrete error which we are working around for each test?

**Comment by SkorikSergey** [comment]
crw-ci-test

**Comment by SkorikSergey** [comment]
ci-test

**Review by dmytro-ndp — APPROVED** [review]
Looks good.

**Pull request merged**